Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py

```diff
@@ -47,14 +46,29 @@ def _get_result(self, test_url):
             _log.debug("Hopefully the benchmark profile has finished writing to disk, moving on.")
         return result
 
-    def _run_one_test(self, web_root, test_file):
+    def _run_one_test(self, web_root, test_file, iteration):
+        enable_profiling = True if self._profile_output_dir else False
+        profile_filename = "{}/{}-{}".format(self._profile_output_dir, self._plan_name, iteration)
         try:
             self._http_server_driver.serve(web_root)
             url = urljoin(self._http_server_driver.base_url(), self._plan_name + '/' + test_file)
-            self._browser_driver.launch_url(url, self._plan['options'], self._build_dir, self._browser_path)
-            timeout = self._plan['timeout']
-            with Timeout(timeout), self._browser_driver.prevent_sleep(timeout):
-                result = self._get_result(url)
+            if enable_profiling:
+                context = self._browser_driver.profile(profile_filename)
+            else:
+                try:
+                    from contextlib import nullcontext
+                except ImportError:
+                    # Python 2 does not have contextlib.nullcontext, so we must define our own.
```

Comment:
Where are we running this script as Python 2?

Reply:
webkitpy EWS has a Python2 component: https://ews-build.webkit.org/#/builders/5/builds/99106/steps/12/logs/stdio